### PR TITLE
Add Java support to RPC API

### PR DESCRIPTION
**Pull request by atomb**
* Adds RPC methods to support Java verification, including to:
    * load a class
    * assume that a method satisfies a contract
    * verify that a method satisfies a contract
* Adds Python client libraries to support Java verification
* Adds a basic test of Java verification functionality
* Adds support for the Java `CLASSPATH` environment variable in both `saw` and `saw-remote-api`

### CODE REVIEW

File: saw-remote-api/saw-remote-api/Main.hs
Comment (atomb) [inline saw-remote-api/saw-remote-api/Main.hs:80]: Oops, I should fill these in!

File: saw-remote-api/python/saw_client/__init__.py
Comment (pnwamk) [inline saw-remote-api/python/saw_client/__init__.py:428]: An optional list is a little funny, given that we treat `None` and `[]` the same. I suppose it's not entirely clear to me if this is a feature or a bug (in both this and `llvm_verify`), hah. I guess leaving it is fine...?

Comment (atomb) [reply saw-remote-api/python/saw_client/__init__.py:428]: Yeah, that's a good question. I didn't write the original version, so I don't know if there was a motivation behind it? The fact that there's a later check for `None` that sets lemmas to `[]` makes it seem eve less necessary.

Comment (atomb) [reply saw-remote-api/python/saw_client/__init__.py:428]: I'm happy to modify both this instance and the one in `llvm_verify` if you don't know of any reason to keep it this way.

File: saw-remote-api/python/tests/saw/test_basic_java.py
Comment (pnwamk) [inline saw-remote-api/python/tests/saw/test_basic_java.py:20]: This test looks like it uses 2 of the 3 added server methods. Is it worth adding another test or modifying this one so it also does a quick sanity check for `SAW/JVM/assume`?

File: saw-remote-api/python/saw_client/__init__.py
Comment (pnwamk) [reply saw-remote-api/python/saw_client/__init__.py:428]: I don't have a strong opinion here - I don't think we strictly need to support both, and it does seem like just using `List` could simplify the code slightly. Your call.

**Review by pnwamk — APPROVED** [review]
LGTM overall. The few things you and I noted (e.g., fill out `TODO`s, consider adding a test for the other function) probably worth knocking out before a merge.

File: saw-remote-api/python/saw_client/__init__.py
Comment (atomb) [reply saw-remote-api/python/saw_client/__init__.py]: Done.

File: saw-remote-api/python/tests/saw/test_basic_java.py
Comment (atomb) [reply saw-remote-api/python/tests/saw/test_basic_java.py:31]: Done.

**Comment by atomb** [comment]
I'm in the process of fixing build issues in a separate PR. Once that's merged, I'll rebase this one and hope it works.

**Pull request merged**
